tests/lib/ocpp/v201/test_profile.cpp

```diff
@@ -0,0 +1,1155 @@
+
+    ASSERT_EQ(expected, actual);
+}
+
+TEST(OCPPTypesTest, CalculateChargingScheduleCombined_Default) {
+    CompositeSchedule expected = {
+        .chargingSchedulePeriod = {ChargingSchedulePeriod{
+            .startPeriod = 0, .limit = DEFAULT_LIMIT_AMPS, .numberPhases = DEFAULT_AND_MAX_NUMBER_PHASES}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = DEFAULT_SCHEDULE.duration,
+        .scheduleStart = DEFAULT_SCHEDULE.scheduleStart,
+        .chargingRateUnit = DEFAULT_SCHEDULE.chargingRateUnit,
+
+    };
+
+    const CompositeSchedule actual =
+        calculate_composite_schedule(DEFAULT_SCHEDULE, DEFAULT_SCHEDULE, DEFAULT_SCHEDULE, DEFAULT_SCHEDULE);
+
+    ASSERT_EQ(expected, actual);
+}
+
+TEST(OCPPTypesTest, CalculateChargingScheduleCombined_CombinedTxDefault) {
+    CompositeSchedule profile = DEFAULT_SCHEDULE;
+    CompositeSchedule tx_default_schedule = {
+        .chargingSchedulePeriod = {{0, 10.0, nullopt}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = 600,
+        .scheduleStart = dt("12:00"),
+        .chargingRateUnit = ChargingRateUnitEnum::A,
+    };
+    CompositeSchedule expected = CompositeSchedule{
+        .chargingSchedulePeriod = {ChargingSchedulePeriod{
+            .startPeriod = 0, .limit = 10, .numberPhases = DEFAULT_AND_MAX_NUMBER_PHASES}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = 600,
+        .scheduleStart = dt("12:00"),
+        .chargingRateUnit = ChargingRateUnitEnum::A,
+    };
+
+    const CompositeSchedule actual = calculate_composite_schedule(profile, profile, tx_default_schedule, profile);
+
+    ASSERT_EQ(expected, actual);
+}
+
+TEST(OCPPTypesTest, CalculateChargingScheduleCombined_CombinedTxDefaultTx) {
+    CompositeSchedule charging_station_max = DEFAULT_SCHEDULE;
+    CompositeSchedule tx_default_schedule = {
+        .chargingSchedulePeriod = {{0, 10.0, nullopt}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = 600,
+        .scheduleStart = dt("12:00"),
+        .chargingRateUnit = ChargingRateUnitEnum::A,
+    };
+    CompositeSchedule tx_schedule = {
+        .chargingSchedulePeriod = {{0, 32.0, nullopt}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = 600,
+        .scheduleStart = dt("12:00"),
+        .chargingRateUnit = ChargingRateUnitEnum::A,
+    };
+
+    CompositeSchedule expected = {
+        .chargingSchedulePeriod = {ChargingSchedulePeriod{
+            .startPeriod = 0, .limit = 32.0, .numberPhases = DEFAULT_AND_MAX_NUMBER_PHASES}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = 600,
+        .scheduleStart = dt("12:00"),
+        .chargingRateUnit = ChargingRateUnitEnum::A,
+    };
+
+    const CompositeSchedule actual =
+        calculate_composite_schedule(DEFAULT_SCHEDULE, charging_station_max, tx_default_schedule, tx_schedule);
+
+    ASSERT_EQ(expected, actual);
+}
+
+TEST(OCPPTypesTest, CalculateChargingScheduleCombined_CombinedOverlapA) {
```

Comment:
Ditto

Reply:
Updated test names to `CalculateChargingScheduleCombined_CombinedOverlapTxAndTxDefault` and `CalculateChargingScheduleCombined_CombinedOverlapTxTxDefaultAndChargingStationMax`.